File: backend/src/main/java/com/app/mevocab/internal/card/dto/request/CardSaveRequest.java
package com.app.mevocab.internal.card.dto.request;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class CardSaveRequest {
    String studentID;
    String word;
    boolean memorized;
    String note;
}

File: frontend/lib/data/models/dto/Request/CardSaveRequest.dart
class CardSaveRequest {
  final String studentID;
  final String word;
  final bool memorized;

  CardSaveRequest({
    required this.studentID,
    required this.word,
    required this.memorized,
  });

  Map<String, dynamic> toJson() {
    return {
      'studentID': studentID,
      'word': word,
      'memorized': memorized,
    };
  }
}
